File: pset6/cash.py
from cs50 import get_float

while True:
    change = get_float("Change Owed: ")
    if change > 0:
        break

# converting change:dollars to cents
value = change * 100
count = 0

# counting of 25 cents
while value >= 25:
    count += 1
    value -= 25

# counting of 10 cents
while value >= 10:
    count += 1
    value -= 10

# counting of 5 cents
while value >= 5:
    count += 1
    value -= 5

# counting of 1 cents
while value >= 1:
    count += 1
    value -= 1

# printing the number of coins
print(count)

File: pset1/cash.c
#include<stdio.h>
#include<math.h>
#include<cs50.h>

//declaring function for counting coins
int count_coins(string prompt);

int main(void)
{
    float val = count_coins("Change owed: ");
    int num = round(val);
    printf("%i\n", num);
}

//writing the function
int count_coins(string prompt)
{
    float change;
    do
    {
        change = get_float("%s", prompt);
    }
    while (change < 0);
    int cents = round(change * 100);
    //intializing count
    int count = 0;

    //now performing counting of coins

    //checking cents are greater than 25 cents
    while (cents >= 25)
    {
        count ++;
        cents = cents - 25;
    }

    //checking cents are greater than 10 cents
    while (cents >= 10)
    {
        count ++;
        cents = cents - 10;
    }

    //checking cents are greater than 5 cents
    while (cents >= 5)
    {
        count ++;
        cents = cents - 5;
    }

    //checking cents are greater than 1 cents
    while (cents >= 1)
    {
        count ++;
        cents = cents - 1;
    }

    return count;
}